Add capability to create and store QR codes in Azure Blob Storage and view in URL Manager

harrykimpel opened the pull request:
This pull request introduces several changes to the URL shortener service, mainly focused on adding support for QR code generation and improving code readability. The most important changes include the addition of a new package, updates to the ShortUrlEntity and ShortResponse classes, and the implementation of QR code generation and storage.

**QR Code Support**:

- [src/Core/Cloud5mins.ShortenerTools.Core.csproj](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-40bf554ecd7d92b42771c774b279965da5a8e54736d6d2c977680edc2ea7ab8bR8): Added Azure.Storage.Blobs package reference to handle QR code image storage.
- [src/Core/Domain/ShortUrlEntity.cs](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-ad6e39eb866e2e312a9163b064dcef8186d532bccb5ad49cfb8b6fd79d9bc2abR28-L32): Added QrCodeUrl property and updated constructors and initialization methods to include this new property. [[1]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-ad6e39eb866e2e312a9163b064dcef8186d532bccb5ad49cfb8b6fd79d9bc2abR28-L32) [[2]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-ad6e39eb866e2e312a9163b064dcef8186d532bccb5ad49cfb8b6fd79d9bc2abL70-R89) [[3]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-ad6e39eb866e2e312a9163b064dcef8186d532bccb5ad49cfb8b6fd79d9bc2abL99-R108)
- [src/Core/Messages/ShortResponse.cs](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c275cbd807a40f7efc5ff15d3ea2a1a66572878fb6815517157215b5765a0aaR8-R23): Added QrCode property and updated constructors to include this new property.
- [src/Core/Service/UrlServices.cs](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c6b5ebdad9d87af8de6435124a96ebb66313aa6d4b65eba626d87bb89c4117fR6): Implemented GetQRCode method to generate QR codes and store them in Azure Blob Storage. Updated Create method to include QR code generation and storage. [[1]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c6b5ebdad9d87af8de6435124a96ebb66313aa6d4b65eba626d87bb89c4117fR6) [[2]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c6b5ebdad9d87af8de6435124a96ebb66313aa6d4b65eba626d87bb89c4117fR109-R112) [[3]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c6b5ebdad9d87af8de6435124a96ebb66313aa6d4b65eba626d87bb89c4117fL120-R127) [[4]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4c6b5ebdad9d87af8de6435124a96ebb66313aa6d4b65eba626d87bb89c4117fR140-R200)

**UI Enhancements**:

- [src/TinyBlazorAdmin/Components/Pages/UrlManager.razor](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4a86b26fece1b7a5142076acd3fa783b017e60fc1a7484f2553e24eb8f9f6e7fL20-R66): Added a new column to display QR codes in the URL manager grid and improved code readability by formatting and organizing the code. [[1]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4a86b26fece1b7a5142076acd3fa783b017e60fc1a7484f2553e24eb8f9f6e7fL20-R66) [[2]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4a86b26fece1b7a5142076acd3fa783b017e60fc1a7484f2553e24eb8f9f6e7fL121-R134) [[3]](https://github.com/microsoft/AzUrlShortener/pull/563/files#diff-4a86b26fece1b7a5142076acd3fa783b017e60fc1a7484f2553e24eb8f9f6e7fL172-R186)

FBoucher commented:
This looks very nice @harrykimpel !
Why using a different storage? could the urlData one could be used instead?

harrykimpel commented:
> This looks very nice @harrykimpel ! Why using a different storage? could the urlData one could be used instead?

that should be possible as well.

fboucheros commented:
Let's me know if you need help. I believe you need to add a blob client (like I did for the table) and pass it to the webapp.

harrykimpel commented:
@fboucheros I removed the unneccessary additional storage account and use the existing url-data storage account. I updated the code to use a blob client, adding into the app host and using it in the UrlServices.cs.

FBoucher commented:
It looks awesome @harrykimpel but in your PR you are using an external service (https://goqr.me/api/) to generate the QRCode. I'm would prefer not having external service dependencies.
Maybe we can find a package that does that?
Otherwise, I really like the idea and the code looks great.

harrykimpel commented:
Thanks @FBoucher!

Let me have a look at some packages that can do that.
